src/util.ml

```diff
@@ -16,13 +18,56 @@ let get_lexing_position lexbuf =
   let column = p.Lexing.pos_cnum - p.Lexing.pos_bol + 1 in
   (line_number, column)
 
+(* Modification of Bytes.escaped to leave UTF-8 bytes unescaped *)
+let escape_bytes s =
+  let n = ref 0 in
+  for i = 0 to B.length s - 1 do
+    n := !n +
+      (match B.unsafe_get s i with
+       | '\"' | '\\' | '\n' | '\t' | '\r' | '\b' -> 2
+       | ' ' .. '~' -> 1
+       | c when (char_code c > 127) -> 1
+       | _ -> 4)
+  done;
+  if !n = B.length s then B.copy s else begin
+    let s' = B.create !n in
+    n := 0;
+    for i = 0 to B.length s - 1 do
+      begin match B.unsafe_get s i with
+      | ('\"' | '\\') as c ->
+          B.unsafe_set s' !n '\\'; incr n; B.unsafe_set s' !n c
+      | '\n' ->
+          B.unsafe_set s' !n '\\'; incr n; B.unsafe_set s' !n 'n'
+      | '\t' ->
+          B.unsafe_set s' !n '\\'; incr n; B.unsafe_set s' !n 't'
+      | '\r' ->
+          B.unsafe_set s' !n '\\'; incr n; B.unsafe_set s' !n 'r'
+      | '\b' ->
+          B.unsafe_set s' !n '\\'; incr n; B.unsafe_set s' !n 'b'
+      | (' ' .. '~') as c -> B.unsafe_set s' !n c
+      | c when (char_code c > 127) -> B.unsafe_set s' !n c
+      | c ->
+          let a = char_code c in
+          B.unsafe_set s' !n '\\';
+          incr n;
+          B.unsafe_set s' !n (char_chr (48 + a / 100));
```

Comment:
I think we should move `48` to a constant with some descriptive name to make the code easier to read for someone who doesn't remember the details of the encoding offhand.

Reply:
I've tried to leave the function as close to the stdlib version as possible to aid readability by comparison, mutatis mutandis, but we can consider this suggestion.